grouper/repositories/interfaces.py

```diff
@@ -44,3 +64,64 @@ def permissions_for_user(self, user):
     def user_has_permission(self, user, permission):
         # type: (str, str) -> bool
         pass
+
+
+class ServiceAccountRepository(object):
+    """Abstract base class for service account repositories."""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def assign_service_account_to_group(self, username, groupname):
+        # type: (str, str) -> None
+        pass
+
+    @abstractmethod
+    def enable_service_account(self, name):
+        # type: (str) -> None
+        pass
+
+    @abstractmethod
+    def set_service_account_description(self, name, description):
+        # type: (str, str) -> None
+        pass
+
+    @abstractmethod
+    def set_service_account_mdbset(self, name, mdbset):
+        # type: (str, str) -> None
+        pass
+
+
+class UserRepository(object):
+    """Abstract base class for user repositories."""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def add_user_to_group(self, username, groupname, role):
+        # type: (str, str, str) -> None
+        pass
+
+    @abstractmethod
+    def disable_user(self, username):
+        # type: (str) -> None
+        pass
+
+    @abstractmethod
+    def enable_user(self, username):
+        # type: (str) -> None
+        pass
+
+    @abstractmethod
```

Comment:
Per above, I'd get rid of this method on the user repository entirely and have the service call the group edge repository for this information.